Apple/NetworkExtension/PacketTunnelProvider.swift

```diff
@@ -1,39 +1,63 @@
 import libburrow
 import NetworkExtension
-import OSLog
+import os
 
 class PacketTunnelProvider: NEPacketTunnelProvider {
-    let logger = Logger(subsystem: "com.hackclub.burrow", category: "General")
+    let logger = Logger(subsystem: "com.hackclub.burrow", category: "frontend")
+    var client: BurrowIpc?
+    var osInitialized = false
     override func startTunnel(options: [String: NSObject]?, completionHandler: @escaping (Error?) -> Void) {
-        let fild = libburrow.retrieve()
-        if fild == -1 {
-            // Not sure if this is the right way to return an error
-            logger.error("Failed to retrieve file descriptor for burrow.")
-            let err = NSError(
-                domain: "com.hackclub.burrow",
-                code: 1_010,
-                userInfo: [NSLocalizedDescriptionKey: "Failed to find TunInterface"]
-            )
-            completionHandler(err)
+        logger.log("Starting tunnel")
+        if !osInitialized {
+            libburrow.initialize_oslog()
+            osInitialized = true
+        }
+        libburrow.start_srv()
+        client = BurrowIpc(logger: logger)
+        logger.info("Started server")
+        Task {
+            do {
+                let command = BurrowRequest(id: 0, command: "ServerConfig")
+                guard let data = try await client?.request(command, type: Response<BurrowResult<ServerConfigData>>.self)
+                else {
+                    throw BurrowError.cantParseResult
+                }
+                let encoded = try JSONEncoder().encode(data.result)
+                self.logger.log("Received final data: \(String(decoding: encoded, as: UTF8.self))")
+                guard let serverconfig = data.result.Ok else {
+                    throw BurrowError.resultIsError
+                }
+                guard let tunNs = self.generateTunSettings(from: serverconfig) else {
+                    throw BurrowError.addrDoesntExist
+                }
+                try await self.setTunnelNetworkSettings(tunNs)
+                self.logger.info("Set remote tunnel address to \(tunNs.tunnelRemoteAddress)")
+                completionHandler(nil)
+            } catch {
+                self.logger.error("An error occurred: \(error)")
+                completionHandler(error)
+            }
         }
-        logger.info("fd: \(fild)")
-        completionHandler(nil)
     }
-
+    private func generateTunSettings(from: ServerConfigData) -> NETunnelNetworkSettings? {
+        let cfig = from.ServerConfig
+        guard let addr = cfig.address else {
+            return nil
+        }
+        let nst = NEPacketTunnelNetworkSettings(tunnelRemoteAddress: addr)
```

Comment:
addr should go in `NEPacketTunnelNetworkSettings.ipv4Settings.addresses` instead

`tunnelRemoteAddress` is intended to be the IP address of the server

Reply:
I see! Currently I've updated this part to:

```swift
var nst = NEPacketTunnelNetworkSettings(tunnelRemoteAddress: "1.1.1.1")
nst.ipv4Settings = NEIPv4Settings(addresses: [addr], subnetMasks: ["0.0.0.0"])
```

Seeing that we will be handling the network interface after its initialization on the daemon side, I suppose the `tunnelRemoteAddress` field of this would not influence any behavior of the program?  The program will error when no `tunnelRemoteAddress` is specified, so I put a makeshift `1.1.1.1` there. 

Also, I'm wondering if we should also send the `TunInterface.netmask()` information to the frontend for ipv4Settings initialization.